content/automate/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/dialog_parser.rb

```diff
@@ -46,6 +46,14 @@ def add_password_value(sequence, option_key, value, options_hash)
   prefixed_option_key = 'password::dialog_' + option_key
   add_hash_value(sequence, stripped_option_key.to_sym, value, options_hash)
   add_hash_value(sequence, prefixed_option_key.to_sym, value, options_hash)
+
+  if special_password?(option_key.to_sym)
+    add_hash_value(sequence, option_key.to_sym, value, options_hash)
+  end
+end
+
+def special_password?(option_key)
+  [:root_password, :sysprep_password, :sysprep_domain_password, :sysprep_admin_password].include?(option_key) ? true : false
```

Comment:
@branic can you make this as a constant and freeze at the top of the file and then use it here.